charmcraft/metafiles/actions.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Charmcraft project handle actions.yaml file."""
+
+import pathlib
+import logging
+from typing import Any, Dict, Optional
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def create_actions(
```

Comment:
Are we going to have to do more with actions later? (e.g. validate them, lint them, etc.?) If so, we should probably go ahead and make a model for this, even if it's very basic right now.

Even if we don't make a full model, the `actions` dict is probably a `Dict[ActionName, pydantic.json_schema.JsonSchemaValue]`, with `ActionName` being either just `str` or a constrained string that does some basic validation (the regex `[A-z_-][A-z0-9_-]*` meets the documentation's requirements).

Reply:
If you only want to do sanity check that can be done. As for now it will check if it is a valid yaml.

Reply:
I added some sanity check for the actions name, but not full model.